src/builder/commands/copy.rs

```diff
@@ -104,6 +115,13 @@ impl BuildStep for Copy {
             // inside the container which is ugly
         }
         hash.field("path", self.path.to_str().unwrap());
+        if let Some(uid) = self.owner_uid {
+            hash.field("owner_uid", uid.to_string());
+        }
+        if let Some(gid) = self.owner_gid {
+            hash.field("owner_gid", gid.to_string());
+        }
+        hash.field("umask", format!("{:o}", self.umask));
```

Comment:
I don't think octal formatting is useful for hashing. I think we will get rid of `.to_string()` in `hash.field` and having to `format!` will make that optimization harder.